File: TMZR Cocoa Touch Class.xctemplate/UIViewControllerObjective-C/___FILEBASENAME___.m
//___FILEHEADER___
//  Copyright <#Organization#> ___YEAR___. All rights reserved.
//

#pragma mark Imports
#import "___FILEBASENAME___.h"

#pragma mark Local defines & static variables


#pragma mark -
#pragma mark Private interface (usable only within class)
@interface ___FILEBASENAMEASIDENTIFIER___() {
}

#pragma mark IBOutlet properties


#pragma mark Other properties


#pragma mark Class methods


#pragma mark Instance methods


@end


#pragma mark -
#pragma mark Implementation
@implementation ___FILEBASENAMEASIDENTIFIER___

#pragma mark Class methods and overwritten methods


#pragma mark Property getter and setter


#pragma mark View lifecycle
/**
 *	Performs the following actions:
 *	- <#Description#>
 */
- (void)viewDidLoad {
	[super viewDidLoad];

	// Custom actions

}

/**
 *	Performs the following actions:
 *	- <#Description#>
 *
 *	@param animated Appearance animation state
 */
- (void)viewWillAppear:(BOOL)animated {
	[super viewWillAppear:animated];

	// Custom actions

}

/**
 *	Performs the following actions:
 *	- <#Description#>
 *
 *	@param animated Appearance animation state
 */
- (void)viewDidAppear:(BOOL)animated {
	[super viewDidAppear:animated];

	// Custom actions

}

/**
 *	Performs the following actions:
 *	- <#Description#>
 *
 *	@param animated Disappearance animation state
 */
- (void)viewWillDisappear:(BOOL)animated {
	[super viewWillDisappear:animated];

	// Custom actions

}

/**
 *	Performs the following actions:
 *	- <#Description#>
 *
 *	@param animated Disappearance animation state
 */
- (void)viewDidDisappear:(BOOL)animated {
	[super viewDidDisappear:animated];

	// Custom actions

}

#pragma mark Navigation
/**
 *	Performs the following actions subject to segue's identifier:
 *	- <#List segues#>
 *
 *	@param segue Segue about to be performed
 *	@param sender Object initiating the segue
 */
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
	// Get the new view controller using [segue destinationViewController]

}

#pragma mark Interface changes
- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
	[super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];

	[coordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext> context) {
	} completion:^(id<UIViewControllerTransitionCoordinatorContext> context) {
		UIInterfaceOrientation orientation = [UIApplication sharedApplication].statusBarOrientation;
		[self adaptUIToOrientation:orientation];
	}];
}

/**
 *	Performs user interface adaptions on orientation changes.
 *
 *	@param orientation Current interface orientation
 */
- (void)adaptUIToOrientation:(UIInterfaceOrientation)orientation {
	// Only uncomment when using MZIK library
	// [super adaptUIToOrientation:orientation];

	// Interface adaptions

}

#pragma mark Memory management
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: TMZR Cocoa Touch Class.xctemplate/UIViewControllerSwift/___FILEBASENAME___.swift
//___FILEHEADER___

// MARK: Imports
import UIKit

// MARK: -
// MARK: Implementation
/**
<#Controller Description#>
*/
class ___FILEBASENAMEASIDENTIFIER___: ___VARIABLE_cocoaTouchSubclass___ {
	// MARK: Nested types


	// MARK: Type properties


	// MARK: Type methods


	// MARK: -
	// MARK: Instance properties


	// MARK: -
	// MARK: View lifecycle
	/**
	Performs the following actions:
	- Sets up the controllers view
	- <#Description#>
	*/
	override func viewDidLoad() {
		super.viewDidLoad()
		
		self.initContent()
		self.setAccessibilityIdentifiers()
	}
	
	/**
	Performs the following actions:
	- <#Description#>
	
	- parameters:
		- animated: Appearance animation state
	*/
	override func viewWillAppear(_ animated: Bool) {
		super.viewWillAppear(animated)
		
		// Custom actions
	}
	
	/**
	Performs the following actions:
	- <#Description#>
	
	- parameters:
		- animated: Appearance animation state
	*/
	override func viewDidAppear(_ animated: Bool) {
		super.viewDidAppear(animated)
		
		// Custom actions
	}
	
	/**
	Performs the following actions:
	- <#Description#>
	
	- parameters:
		- animated: Disappearance animation state
	*/
	override func viewWillDisappear(_ animated: Bool) {
		super.viewWillDisappear(animated)
		
		// Custom actions
	}
	
	/**
	Performs the following actions:
	- <#Description#>
	
	- parameters:
		- animated: Disappearance animation state
	*/
	override func viewDidDisappear(_ animated: Bool) {
		super.viewDidDisappear(animated)
		
		// Custom actions
	}

	// MARK: -
	// MARK: Instance methods
	
}

// MARK: -
// MARK: Navigation
extension ___FILEBASENAMEASIDENTIFIER___ {
	override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
		// Get the new view controller using segue.destinationViewController.
		
	}
}

// MARK: -
// MARK: Setup and initialization
extension ___FILEBASENAMEASIDENTIFIER___ {
	/**
	Initilalizes the content of the controller's view.
	*/
	func initContent() {
		self.styleContent()
		self.initStaticContent()
		self.initDynamicContent()
	}
	
	/**
	Styles the content of the controller's view.
	*/
	func styleContent() {
		
	}
	
	/**
	Initializes the static content of the controller's view subject to the current locale.
	*/
	private func initStaticContent() {
		
	}
	
	/**
	Initializes the dynamic content of the controller's view.
	*/
	private func initDynamicContent() {
		
	}
}

// MARK: -
// MARK: Orientation
extension ___FILEBASENAMEASIDENTIFIER___ {
	override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
		super.viewWillTransition(to: size, with: coordinator)
		
		coordinator.animate(alongsideTransition: nil) { [weak self] (context) in
			// Put custom transition animations here
		}
	}
}

// MARK: -
// MARK: Accessibility
extension ___FILEBASENAMEASIDENTIFIER___ {
	func setAccessibilityIdentifiers() {
		
	}
}
